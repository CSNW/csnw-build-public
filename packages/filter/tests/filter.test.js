const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const filter = require('../')('!b.js');
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await filter.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*.{js,html}', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
