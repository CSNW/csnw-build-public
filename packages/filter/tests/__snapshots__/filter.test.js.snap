// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/filter/tests/fixtures",
    "contents": "var name = 'a';
",
    "cwd": "C:/dev/csnw-build/packages/filter/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/filter/tests/fixtures/a.js",
    "sourceMap": null,
  },
  "c.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/filter/tests/fixtures",
    "contents": "var name = 'c';
",
    "cwd": "C:/dev/csnw-build/packages/filter/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/filter/tests/fixtures/c.js",
    "sourceMap": null,
  },
}
`;
